source/common/common/token_bucket_impl.cc

```diff
@@ -17,21 +17,25 @@ bool TokenBucketImpl::consume(uint64_t tokens) {
     last_fill_ = time_now;
   }
 
+  if (allow_partial) {
+    tokens = std::min(tokens, static_cast<uint64_t>(std::floor(tokens_)));
+  }
+
   if (tokens_ < tokens) {
     return false;
```

Comment:
nit: should we return zero here to be consistent with return value?

Reply:
Yup will fix.